Cv tooltip design docs (#1326)
hashicorp/design-system: website/docs/components/tooltip/partials/guidelines/guidelines.md

### When to use

> When displaying additional context or information that is not critical to the user's task or decision-making process.

Reviewer: ```suggestion
When displaying additional context or information that is not critical to the user’s task or decision-making process.
```

### When not to use

- When the content is too lengthy or complex.

> - When the information provided is critical to the user's task or decision-making process, consider moving the content to the page or an [alert](https://helios.hashicorp.design/components/alert).

Reviewer: ```suggestion
- When the information provided is critical to the user’s task or decision-making process, consider moving the content to the page or an [alert](https://helios.hashicorp.design/components/alert).
```

> tooltip-when-to-use

Author: Yes 🤦 Fixed!

!!! Do

![Correct use of a tooltip](/assets/components/tooltip/tooltip-when-to-use.png)

!!!

> !!! Dont

Reviewer 2: [non-blocking] It seems like it's helpful in the do/dont blocks to specifically call out what the recommendation is. Specifically in this scenario it could be interpreted in a couple of different ways: Don't use a tooltip in place of helper/placeholder text, don't use a tooltip on form fields, don't use a tooltip to highlight blocking actions, etc. Probably good for us to align holistically on what goes inside of these blocks other than images/examples.

Author: Thanks for the feedback, Jory. This is a good point. I added a short sentence to each block, but I'm unsure if the sentence in the Don't block sounds good or a little bit forced. Wdyt?

`Avoid using tooltips to place crucial information related to the user's task.`

Reviewer 2: Yep, that's feels good to me and is explicit enough

![Incorrect use of a tooltip](/assets/components/tooltip/tooltip-when-not-to-use.png)

!!!

## Placement

Tooltips can be placed in various positions relative to the UI element they are associated with. Choose the placement based on the context and available space around the tooltip trigger.

There are twelve placement options: `top`, `bottom`, `left`, `right`, `top-start`, `top-end`, `bottom-start`, `bottom-end`, `left-start`, `left-end`, `right-start`, `right-end`.

> 

Reviewer: I think it's confusing to say there are twelve placement options but then show a picture of only four.

I would change the text to:

"There are four main placement options: `top`, `bottom`, `left`, and `right`."

You _could_ mention that there are other placement options as well but, personally, I don't think the other options are really useful or necessary and I think listing them all is likely to be more confusing than helpful.

![Tooltip with wrapped text](/assets/components/tooltip/tooltip-placement.png)

## Offset

The default and recommended distance between the trigger and the tooltip's pointer or arrow is 4px.

![Default offset of the tooltip](/assets/components/tooltip/tooltip-spacing.png)

!!! Info

>  In some cases, an offset may be necessary to adjust the position of the tooltip. The offset should be used sparingly when it's necessary to make sure that the tooltip does not obscure or cover the trigger or important information.

Reviewer: ```suggestion
 In some cases, an offset may be necessary to adjust the position of the tooltip. The offset should be used sparingly when it’s necessary to make sure that the tooltip does not obscure or cover the trigger or important information.
```

Author: Thanks, Kristin. I love this improved version!

!!!

!!! Do

> Adjust the offset when the tooltip needs to be positioned in a way that it doesn't obstruct the view of the UI element or information it is associated with.

Reviewer: ```suggestion
Adjust the offset when the tooltip needs to be positioned in a way that it doesn’t obstruct the view of the UI element or information it is associated with.
```

![Tooltip with adjusted offset](/assets/components/tooltip/tooltip-offset-do.png)
!!!

!!! Dont

> Don't add extra offset if the tooltip blocks important information or appears disjointed from the element it is meant to provide information for, and can cause confusion for the user. 

Reviewer: ```suggestion
Don’t add extra offset if the tooltip would block important information, appear disjointed from the element it is meant to provide information for, or cause confusion for the user.
```

![Tooltip with adjusted offset](/assets/components/tooltip/tooltip-offset-dont.png)
!!!

## Content

> The tooltip should contain text only. Basic text formatting, such as bold and italic is supported.

Reviewer: ```suggestion
The tooltip should contain text only. Basic text formatting, such as bold and italic, is supported.
```

If more complex content is necessary to convey the information, consider other display components or moving the content to the page.

![Tooltip with bold and italic text](/assets/components/tooltip/tooltip-text-formatting.png)

## Text overflow

> The text should wrap when it exceeds a maximum width of 280px. There is no character limit, but we recommend keeping them short and concise (~80 characters).

Reviewer: ```suggestion
The text should wrap when it exceeds a maximum width of 280px. There is no character limit, but we recommend keeping it short and concise (~80 characters).
```

![Tooltip with wrapped text](/assets/components/tooltip/tooltip-text-wrapping.png)

!!! Info

> Text wrapping can be achieved through the property `isMultiline` in Figma.

Reviewer: ```suggestion
Text wrapping can be achieved using the property `isMultiline` in Figma.
```
